src/client.ts

```diff
@@ -212,12 +214,10 @@ export interface ICreateClientOpts {
     scheduler?: MatrixScheduler;
 
     /**
-     * The function to invoke for HTTP
-     * requests. The value of this property is typically <code>require("request")
-     * </code> as it returns a function which meets the required interface. See
-     * {@link requestFunction} for more information.
+     * The function to invoke for HTTP requests.
+     * See {@link fetch} for more information.
      */
-    request?: IHttpOpts["request"];
+    fetch?: typeof fetch;
```

Comment:
I think maybe instead of depending on cross-fetch we could just have the caller pass their favourite fetch, as for typing a shim in `/@types/` would be fine